### ISSUE TITLE
Expose an API to create tagged hashes from string?

### ISSUE THREAD

**Issue by sanket1729** [rust-bitcoin/bitcoin_hashes#115]
I see that the current preferred way in the library to create tagged hashes is to `Impl Tag` on the struct. Would it be useful to add a simple method that takes in String and outputs a BIP340 tagged HashEngine. 

Currently, it seems there is some code bloat when what in almost all cases what we need is 

`let hasher = HashEngine::from_bip340tag(&str)`;

This is inefficient compared to precomputing the Midstate, but is really handy for more general applications of bip340 tagged hashes and some quick prototyping. 

If there are no Nacks, I can add a PR to do this.

**Comment by stevenroose** [comment]
Yeah I think it can be handy to have a general function. Though I think it's also important to optimize the common hashes (for taproot stuff) as much as we can.

**Comment by sgeisler** [comment]
I think this is a good idea, maybe point out the optimization option in the docs. Maybe an optional dependency on `lazy_static` ([or implementing something similar by hand as done in `secp256k1`](https://github.com/rust-bitcoin/rust-secp256k1/blob/master/src/context.rs#L25)) would help here, providing an easily cloneable hash engine for common prefixes.

**Comment by dr-orlovsky** [comment]
FYI I did a structure for generating dynamically-tagged hashes here: https://github.com/LNP-BP/client_side_validation/blob/master/commit_verify/src/tagged_hash.rs
